src/analysis/normed/group/pointwise.lean

```diff
@@ -48,46 +48,157 @@ begin
   ... ≤ Rs + Rt : add_le_add (hRs x hx) (hRt y hy)
 end
 
-@[simp] lemma singleton_add_ball (x y : E) (r : ℝ) :
-  {x} + ball y r = ball (x + y) r :=
-by simp only [preimage_add_ball, image_add_left, singleton_add, sub_neg_eq_add, add_comm y x]
+variables {ε δ : ℝ} {s : set E}
 
-@[simp] lemma ball_add_singleton (x y : E) (r : ℝ) :
-  ball x r + {y} = ball (x + y) r :=
-by simp [add_comm _ {y}, add_comm y]
+lemma ennreal.of_real_of_nonpos {r : ℝ} (hr : r ≤ 0) : ennreal.of_real r = 0 :=
+congr_arg coe $ real.to_nnreal_of_nonpos hr
 
-lemma singleton_add_ball_zero (x : E) (r : ℝ) :
-  {x} + ball 0 r = ball x r :=
-by simp
+lemma thickening_of_nonpos (hδ : δ ≤ 0) (s : set E) : thickening δ s = ∅ :=
+eq_empty_iff_forall_not_mem.2 $ λ x hx,
+  (ennreal.of_real_of_nonpos hδ).not_gt $ bot_le.trans_lt hx
 
-lemma ball_zero_add_singleton (x : E) (r : ℝ) :
-  ball 0 r + {x} = ball x r :=
-by simp
+@[simp] lemma thickening_thickening_subset (ε δ : ℝ) (s : set E) :
+  thickening ε (thickening δ s) ⊆ thickening (ε + δ) s :=
+begin
+  intro x,
+  simp_rw mem_thickening_iff,
+  rintro ⟨y, ⟨z, hz, hyz⟩, hxy⟩,
+  exact ⟨z, hz, (dist_triangle _ _ _).trans_lt $ add_lt_add hxy hyz⟩,
+end
```

Comment:
```suggestion
@[simp] lemma thickening_thickening_subset {α : Type*} [pseudo_emetric_space α] (ε δ : ℝ) (s : set α) :
  thickening ε (thickening δ s) ⊆ thickening (ε + δ) s :=
begin
  by_cases ε_sign : ε ≤ 0,
  { simp only [thickening_of_nonpos' ε_sign, empty_subset], },
  by_cases δ_sign : δ ≤ 0,
  { simp only [thickening_of_nonpos' δ_sign, thickening_empty, empty_subset], },
  simp only [not_le] at ε_sign δ_sign,
  intros x,
  simp_rw mem_thickening_iff_exists_edist_lt,
  rintro ⟨y, ⟨z, hz, hyz⟩, hxy⟩,
  refine ⟨z, hz, _⟩,
  apply lt_of_le_of_lt (edist_triangle x y z) (lt_of_lt_of_le (ennreal.add_lt_add hxy hyz) _),
  simp only [ennreal.of_real_add ε_sign.le δ_sign.le, le_refl],
end
```
Again, this works in the generality in which thickenings are defined, and the same (I believe) for the next few lemmas. The proofs unfortunately become clumsier because the ennreals seem to necessitate splitting to cases according to some signs, but I think the lemmas should exist in the right generality.